src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -120,6 +120,24 @@ static void create_static_function_call(
     call.arguments()[0].make_typecast(need_type);
 }
 
+/// Create a disjunction, or add to an existing one. For example,
+/// `add_to_disjunction(x || y, z)` = `x || y || z` (a ternary operation, not
+/// nested binary operations), but `add_to_disjunction(w, z)` where `w` is not
+/// an or_exprt already = `w || z`.
+/// \param maybe_or_expr: expression to add to (`x || y` or `w` in the examples)
+/// \param new_disjunct: new disjunct, written `z` in the examples.
+static void add_to_disjunction(exprt &maybe_or_expr, const exprt &new_disjunct)
```

Comment:
I'd just do `or_exprt(maybe_or_expr, new_disjunct)` and leave the rest to the simplifier. Unless you have data that suggests you need to optimise in exactly this way right here.